build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -0,0 +1,264 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from "fs";
+import { EOL as newline } from "os";
+import * as childProcess from "child_process";
+// eslint-disable-next-line unicorn/import-style
+import * as path from "path";
+import { Flags } from "@oclif/core";
+import { policyHandlers, readJsonAsync } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+
+const readStdin: () => Promise<string | undefined> = async () => {
+    return new Promise((resolve) => {
+        const stdin = process.openStdin();
+        stdin.setEncoding("utf-8");
+
+        let data = "";
+        stdin.on("data", (chunk) => {
+            data += chunk;
+        });
+
+        stdin.on("end", () => {
+            resolve(data);
+        });
+
+        if (stdin.isTTY) {
+            resolve("");
+        }
+    });
+};
+
+type policyAction = "handle" | "resolve" | "final";
+
+export class CheckPolicy extends BaseCommand<typeof CheckPolicy.flags> {
```

Comment:
Can you add some TSDoc comments to the class and functions? Imagine you have to do this again in 6 months -- what do you wish you'd known?